File: RuneSharper.Shared/Models/CharacterViewModel.cs
using RuneSharper.Shared.Enums;

namespace RuneSharper.Shared.Models;

public class CharacterViewModel
{
    public string UserName { get; set; } = string.Empty;
    public DateTime FirstTracked { get; set; }
    public StatsModel? Stats { get; set; }
    public ActivitiesModel? Activities { get; set; }
    public IEnumerable<ActivitiesChangeModel> ActivitiesChange { get; set; } = Enumerable.Empty<ActivitiesChangeModel>();
    public IEnumerable<StatsChangeModel> StatsChange { get; set; } = Enumerable.Empty<StatsChangeModel>();
}

File: RuneSharper.Client/src/app/models/character-view.model.ts
import { Activities, ActivitiesChange } from "./activities.model";
import { Stats, StatsChange } from "./stats.model";

export interface CharacterView {
    userName: string;
    firstTracked: Date;
    stats: Stats;
    activities: Activities;
    activitiesChange: ActivitiesChange[];
    statsChange: StatsChange[];
}